src/BenchmarkDotNet.Toolchains.Uap/Toolchains/Uap/DevicePortalApiWrapper.cs

```diff
@@ -0,0 +1,314 @@
+
+            i = 10;
+            PackageStruct ret;
+            while ((ret = client.Get<PackagesStruct>(packagesRequest).Data.InstalledPackages.SingleOrDefault(x => x.Name.Contains("BenchmarkDotNet"))) == null)
+            {
+                if (--i == 0)
+                {
+                    throw new InvalidOperationException("Installation failed");
+                }
+            }
+
+            return ret;
+        }
+
+        public void RunApplication(PackageStruct id)
+        {
+            var executeRequest = new RestRequest("api/taskmanager/app", Method.POST);
+
+            executeRequest.AddHeader("X-CSRF-Token", csrfToken.Value);
+            executeRequest.AddCookie(csrfToken.Name, csrfToken.Value);
+            executeRequest.AddCookie(wmid.Name, wmid.Value);
+            executeRequest.AddQueryParameter("appid", Convert.ToBase64String(Encoding.UTF8.GetBytes(id.PackageRelativeId)));
+            {
+                var response = client.Post<ProgressStruct>(executeRequest);
+                if (response.StatusCode != HttpStatusCode.OK)
+                {
+                    throw new InvalidOperationException("Execute request failed: " + response.Data.CodeText + " " + response.Data.Reason);
+                }
+            }
+
+            for (;;)
+            {
+                var processesRequest = new RestRequest("/api/resourcemanager/processes", Method.GET);
+
+                processesRequest.AddHeader("X-CSRF-Token", csrfToken.Value);
+                processesRequest.AddCookie(csrfToken.Name, csrfToken.Value);
+                processesRequest.AddCookie(wmid.Name, wmid.Value);
+                {
+                    var response = client.Execute(processesRequest);
+                    if (!response.Content.Contains("BenchmarkDotNet"))
+                    {
+                        break;
+                    }
+
+                    Thread.Sleep(100);
+                }
+            }
+        }
+
+        public void DeleteApplication(PackageStruct appId)
+        {
+            var deleteRequest = new RestRequest("api/app/packagemanager/package", Method.DELETE);
+            deleteRequest.AddQueryParameter("package", appId.PackageFullName);
+
+            deleteRequest.AddHeader("X-CSRF-Token", csrfToken.Value);
+            deleteRequest.AddCookie(csrfToken.Name, csrfToken.Value);
+            deleteRequest.AddCookie(wmid.Name, wmid.Value);
+
+            HttpStatusCode status = client.Execute(deleteRequest).StatusCode;
+            if (status != HttpStatusCode.OK)
+            {
+                throw new InvalidOperationException("Failed to delete package: " + status);
+            }
+        }
+
+        private static long GetTime()
+        {
+            long retval = 0;
+            var st = new DateTime(1970, 1, 1);
+            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
+            retval = (long)(t.TotalMilliseconds + 0.5);
+            return retval;
+        }
```

Comment:
OK, I've figured this out. It's replication of JQuery logic, that adds additional parameter "_" to force servers to response with not cached data. I think this could be removed.